src/extensions.rs

```diff
@@ -40,19 +43,25 @@ mod tests {
         assert_eq!(ext, ext2);
     }
 
+    // todo: maybe make a macro for this?
     #[test]
     fn test_extension_serde() {
         roundtrip_extensions(Extensions::default());
         roundtrip_extensions(Extensions::UNWRAP_NEWTYPES);
         roundtrip_extensions(Extensions::IMPLICIT_SOME);
         roundtrip_extensions(Extensions::UNWRAP_VARIANT_NEWTYPES);
+        roundtrip_extensions(Extensions::EXPLICIT_STRUCT_NAMES);
         roundtrip_extensions(Extensions::UNWRAP_NEWTYPES | Extensions::IMPLICIT_SOME);
         roundtrip_extensions(Extensions::UNWRAP_NEWTYPES | Extensions::UNWRAP_VARIANT_NEWTYPES);
+        roundtrip_extensions(Extensions::UNWRAP_NEWTYPES | Extensions::EXPLICIT_STRUCT_NAMES);
         roundtrip_extensions(Extensions::IMPLICIT_SOME | Extensions::UNWRAP_VARIANT_NEWTYPES);
+        roundtrip_extensions(Extensions::IMPLICIT_SOME | Extensions::EXPLICIT_STRUCT_NAMES);
+        roundtrip_extensions(Extensions::UNWRAP_VARIANT_NEWTYPES | Extensions::EXPLICIT_STRUCT_NAMES);
         roundtrip_extensions(
             Extensions::UNWRAP_NEWTYPES
                 | Extensions::IMPLICIT_SOME
-                | Extensions::UNWRAP_VARIANT_NEWTYPES,
+                | Extensions::UNWRAP_VARIANT_NEWTYPES
+                | Extensions::EXPLICIT_STRUCT_NAMES,
```

Comment:
I'm not sure if i got this right.

Reply:
What we really want to test here is the powerset of all extensions (while that is still possible). Perhaps there is a way to use the bitset API to iterate over the powerset and then invoke the test function for each subset?

Reply:
I've figured out a clean way to do this and implemented it.